src/index.ts

```diff
@@ -66,25 +148,26 @@ function Upload (cfg) {
   this.numBytesWritten = 0;
   this.numRetries = 0;
 
-  const contentLength = cfg.metadata ? parseInt(cfg.metadata.contentLength, 10) : NaN;
+  const contentLength = cfg.metadata ? Number(cfg.metadata.contentLength) : NaN;
   this.contentLength = isNaN(contentLength) ? '*' : contentLength;
 
-  this.once('writing', function () {
-    if (self.uri) {
-      self.continueUploading();
+  this.once('writing', function() {
+    if (this.uri) {
```

Comment:
Intentional? I would have expected you to have changed the anon function to an arrow function when making this change. The one way you can get away without doing that would be if `.once` applies the callback with its own `this`. Does it?

Reply:
You sir are a code review hero.  🥇  This was a bug that alarmingly wasn't caught by tests.  A lot of this gets cleaned up when we go to es6 (which comes after this).

Reply:
I think the event handlers of a stream provide ‘this’ as the context of the steam, so it shouldn’t have caused any problems.